modules/apps/roles/roles-admin-web/src/main/java/com/liferay/roles/admin/web/internal/group/type/contributor/util/GroupTypeContributorUtil.java

```diff
@@ -14,60 +14,33 @@
 
 package com.liferay.roles.admin.web.internal.group.type.contributor.util;
 
-import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.osgi.service.tracker.collections.list.ServiceTrackerList;
+import com.liferay.osgi.service.tracker.collections.list.ServiceTrackerListFactory;
+import com.liferay.petra.function.transform.TransformUtil;
+import com.liferay.portal.kernel.module.util.SystemBundleUtil;
 import com.liferay.portal.kernel.util.PortalUtil;
 import com.liferay.roles.admin.group.type.contributor.GroupTypeContributor;
 
-import java.util.ArrayList;
-import java.util.List;
-
-import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Reference;
-import org.osgi.service.component.annotations.ReferenceCardinality;
-import org.osgi.service.component.annotations.ReferencePolicy;
-import org.osgi.service.component.annotations.ReferencePolicyOption;
-
 /**
  * @author Alejandro Tardín
  */
-@Component(service = {})
 public class GroupTypeContributorUtil {
 
 	public static long[] getClassNameIds() {
-		List<Long> classNameIds = new ArrayList<>();
-
-		for (GroupTypeContributor groupTypeContributor :
-				_groupTypeContributors) {
-
-			if (!groupTypeContributor.isEnabled()) {
-				continue;
-			}
-
-			classNameIds.add(
-				PortalUtil.getClassNameId(groupTypeContributor.getClassName()));
-		}
-
-		return ListUtil.toLongArray(classNameIds, Long::valueOf);
-	}
-
-	@Reference(
-		cardinality = ReferenceCardinality.MULTIPLE,
-		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY
-	)
-	protected void addGroupTypeContributor(
-		GroupTypeContributor groupTypeContributor) {
-
-		_groupTypeContributors.add(groupTypeContributor);
-	}
-
-	protected void removeGroupTypeContributor(
-		GroupTypeContributor groupTypeContributor) {
-
-		_groupTypeContributors.remove(groupTypeContributor);
+		return TransformUtil.transformToLongArray(
+			_serviceTrackerList.toList(),
+			groupTypeContributor -> {
+				if (!groupTypeContributor.isEnabled()) {
+					return null;
+				}
+
+				return PortalUtil.getClassNameId(
+					groupTypeContributor.getClassName());
+			});
 	}
 
-	private static final List<GroupTypeContributor> _groupTypeContributors =
-		new ArrayList<>();
+	private static final ServiceTrackerList<GroupTypeContributor>
+		_serviceTrackerList = ServiceTrackerListFactory.open(
+			SystemBundleUtil.getBundleContext(), GroupTypeContributor.class);
```

Comment:
@JanisZhang Should use `FrameworkUtil.getBundle` and then `Bundle.getBundleContext`. You can initialize it in a static block.